src/workerd/api/web-socket.c++

```diff
@@ -242,9 +242,9 @@ jsg::Ref<WebSocket> WebSocket::constructor(
     headers.unset(kj::HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS);
   }
 
-  auto prom = client->openWebSocket(connUrl, headers)
-      .then([client = kj::mv(client), &context, wsErr = kj::mv(wsErr)]
-            (kj::HttpClient::WebSocketResponse response) mutable -> kj::Promise<PackedWebSocket> {
+  auto prom = ([](auto& context, auto connUrl, auto headers, auto client)
```

Comment:
Instead of a lambda, I think this should be changed to a standalone function (especially since it doesn't capture anything).

Reply:
Given that it is only used here, I'd rather keep it local here. This also minimizes the change delta.